iocore/net/SSLClientUtils.cc

```diff
@@ -66,6 +66,7 @@ verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
   netvc = SSLNetVCAccess(ssl);
 
   if (netvc != nullptr) {
+    netvc->callHooks(TS_EVENT_SSL_SERVER_VERIFY_HOOK);
```

Comment:
Ok, I agree that we should/could move the callback ahead of the depth check.  Not clear that we should bypass the native openssl check.  Do you have a use case for that?

At one point we talked about having different return values for the hook to control whether we also should run the ATS verify checks.